client/widgets/MiscWidgets.cpp

```diff
@@ -253,7 +254,16 @@ void CArmyTooltip::init(const InfoAboutArmy &army)
 		{
 			//if =0 - we have no information about stack size at all
 			if(slot.second.count)
-				subtitle = CGI->generaltexth->arraytxt[171 + 3*(slot.second.count)];
+			{
+				if(settings["adventure"]["numericStackQuantities"].Bool())
+				{
+					subtitle = CCreature::getQuantityRangeStringForId((CCreature::CreatureQuantityId)slot.second.count);
+				}
+				else
+				{
+					subtitle = CGI->generaltexth->arraytxt[171 + 3*(slot.second.count)];
```

Comment:
I see this 171+3* several times in code. What if making map non-const and overriding its values somewhere in the game loading if flag is false? Then you will just to load value from CCreature::getQuantityRangeStringForId whatever is there

Reply:
Not yet sure it will work fine, as I plan ingame toggle for this option somewhere in the future, so displayed way will change instantly

And there are bigger problems with related code, like CArmyInfo (or something that uses it?) sends quantityId 0 from server to indicate that count is "unknown"... etc

Reply:
Another option - move all handling of this text into your method to avoid this duplication, e.g. something like
```
CCreature::getQuantityRangeStringForId()
{
    if (setting{"XXX"].Bool())
        return creatureQuantityRanges[count]; // return numeric form, e.g. 10-20
    else
        return CGI->generaltexth->translate("core.arraytxt", 171 + 3*count); // return H3 text for, e.g. "Few"
}
```

Reply:
but the base offset is not always 171

Reply:
Changing instantly will work too, because you can override map values when toggle this option

Reply:
As I mentioned after, the value varies from 171 to 174, providing extra words that are present in H3, but are not really necessary in concise numerical format.